AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java

```diff
@@ -433,57 +435,56 @@ private void configureToolbarInternal(boolean recur) {
         }
     }
 
-
-    @SuppressWarnings("deprecation") // Tracked as https://github.com/ankidroid/Anki-Android/issues/8293
-    @Override
-    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
-        super.onActivityResult(requestCode, resultCode, intent);
-        Timber.d("onActivityResult (requestCode = %d, resultCode = %d)", requestCode, resultCode);
-
-        // rebuild action bar
+    ActivityResultLauncher<Intent> onRequestPreviewActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
         configureToolbar();
-
-        // boot back to deck picker if there was an error
-        if (resultCode == DeckPicker.RESULT_DB_ERROR || resultCode == DeckPicker.RESULT_MEDIA_EJECTED) {
-            closeStudyOptions(resultCode);
+        if (result.getResultCode() == DeckPicker.RESULT_DB_ERROR || result.getResultCode() == DeckPicker.RESULT_MEDIA_EJECTED) {
+            closeStudyOptions(result.getResultCode());
             return;
         }
-
-        // perform some special actions depending on which activity we're returning from
-        if (requestCode == STATISTICS || requestCode == BROWSE_CARDS) {
-            // select original deck if the statistics or card browser were opened,
-            // which can change the selected deck
-            if (intent.hasExtra("originalDeck")) {
-                getCol().getDecks().select(intent.getLongExtra("originalDeck", 0L));
+        if (result.getResultCode() == Reviewer.RESULT_NO_MORE_CARDS) {
+            // If no more cards getting returned while counts > 0 (due to learn ahead limit) then show a snackbar
+            if (getCol().getSched().count() > 0 && mStudyOptionsView != null) {
```

Comment:
explicit parentheses - these are just nitpicks of course

```suggestion
            if ((getCol().getSched().count() > 0) && (mStudyOptionsView != null)) {
```